### PR TITLE
chore: make `rosbag image pipeline` module generic

### PR DESCRIPTION
**Pull request by malachi-constant**
Make `rosbag-image-pipeline` module generic.

Seedfarmer testing **completed**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by LeonLuttenberger — DISMISSED** [review]

File: modules/analysis/rosbag-image-pipeline/modulestack.yaml [hunk 1]
```diff
@@ -25,25 +28,29 @@ Resources:
...
+              - "arn:aws:s3:::cdk*"
           - Effect: Allow
             Action:
               - "ecr:Describe*"
               - "ecr:Get*"
               - "ecr:List*"
               - "ecr:BatchGetImage"
             Resource:
-              - !Sub "arn:aws:ecr:${AWS::Region}:${AWS::AccountId}:repository/addf-${DeploymentName}-${ModuleName}*"
+              - !Sub "arn:aws:ecr:${AWS::Region}:${AWS::AccountId}:repository/${ProjectName}-${DeploymentName}-${ModuleName}*"
```
Comment (srinivasreddych) [inline modules/analysis/rosbag-image-pipeline/modulestack.yaml:43]: You have ecr:* below. Do you still need these set of permissions?

File: modules/analysis/rosbag-image-pipeline/modulestack.yaml [hunk 2]
```diff
@@ -25,25 +28,29 @@ Resources:
...
           - Effect: Allow
             Action:
               - ecr:*
             Resource:
-              - !Sub "arn:aws:ecr:${AWS::Region}:${AWS::AccountId}:repository/addf-${DeploymentName}-${ModuleName}*"
+              - !Sub "arn:aws:ecr:${AWS::Region}:${AWS::AccountId}:repository/${ProjectName}-${DeploymentName}-${ModuleName}*"
           - Action:
               - "iam:CreateServiceLinkedRole"
```
Comment (srinivasreddych) [inline modules/analysis/rosbag-image-pipeline/modulestack.yaml:50]: Do you need servicelinkedrole permissions here?

**Review by dgraeber — APPROVED** [review]

File: modules/analysis/rosbag-image-pipeline/modulestack.yaml (hunk 1 above)
Comment (malachi-constant) [reply modules/analysis/rosbag-image-pipeline/modulestack.yaml:43]: It's possible its redundant with the the prior permissions, yes. This is from Kevin Soucy's initial commit 2 years ago. Should we test this with removal now or in a separate PR? @dgraeber @srinivasreddych

File: modules/analysis/rosbag-image-pipeline/modulestack.yaml (hunk 2 above)
Comment (malachi-constant) [reply modules/analysis/rosbag-image-pipeline/modulestack.yaml:50]: This is from Kevin Soucy's initial commit 2 years ago. Should we test this with removal now or in a separate PR? @dgraeber @srinivasreddych

Comment (dgraeber) [reply modules/analysis/rosbag-image-pipeline/modulestack.yaml:50]: Not sure why it is even there to be honest

Comment (malachi-constant) [reply modules/analysis/rosbag-image-pipeline/modulestack.yaml:50]: I vote merging this as it fits the scope and then opening a separate issue, to refine the module's permissions

**Review by dgraeber — APPROVED** [review]

File: modules/analysis/rosbag-image-pipeline/modulestack.yaml (hunk 2 above)
Comment (malachi-constant) [reply modules/analysis/rosbag-image-pipeline/modulestack.yaml:50]: https://github.com/awslabs/autonomous-driving-data-framework/issues/503

File: modules/analysis/rosbag-image-pipeline/modulestack.yaml (hunk 1 above)
Comment (malachi-constant) [reply modules/analysis/rosbag-image-pipeline/modulestack.yaml:43]: https://github.com/awslabs/autonomous-driving-data-framework/issues/503

**Pull request merged**